philips-hue-adapter.js

```diff
@@ -103,35 +111,55 @@ class PhilipsHueDevice extends Device {
   /**
    * @param {PhilipsHueAdapter} adapter
    * @param {String} id - A globally unique identifier
-   * @param {String} lightId - id of the light expected by the bridge API
-   * @param {Object} light - the light API object
+   * @param {String} deviceId - id of the device expected by the bridge API
+   * @param {Object} device - the device API object
    */
-  constructor(adapter, id, lightId, light) {
+  constructor(adapter, id, deviceId, device) {
     super(adapter, id);
 
-    this.lightId = lightId;
-    this.name = light.name;
-
-    this.type = THING_TYPE_ON_OFF_LIGHT;
-    this.properties.set('on',
-      new PhilipsHueProperty(this, 'on', {type: 'boolean'}, light.state.on));
+    this.deviceId = deviceId;
+    this.name = device.name;
+
+    if (deviceId.startsWith('sensors')) {
+      if (device.state.hasOwnProperty('presence')) {
+        this.type = Constants.THING_TYPE_BINARY_SENSOR;
+        this.properties.set('on',
+          new PhilipsHueProperty(this, 'on', {type: 'boolean'},
+                                 device.state.presence));
+      } else if (device.state.hasOwnProperty('temperature')) {
+        this.type = Constants.THING_TYPE_UNKNOWN_THING;
```

Comment:
Maybe this would be more appropriate as `THING_TYPE_MULTI_LEVEL_SENSOR`.

Reply:
multiLevelSensor is specified as taking values from 0-100 but the temperature may fall out of that range, especially when it's this time of the year where I live.

Reply:
Right, I forgot we went with percentage instead of a bare number.